m3/data_prepare/report/README.md

```diff
@@ -0,0 +1,69 @@
+  - [3. Convert Data to VLM Model Format](#3-convert-data-to-vlm-model-format)
+
+## Overview
+
+In medical AI applications, particularly for **medical report generation**, preparing data properly is crucial for effective model training. This README automates the preparation of datasets by:
+1. Downloading paired images and medical reports.
+2. Utilizing a Large Language Model (LLM) to refine the text in the reports.
+3. Converting both images and processed text into a format suitable for training the VLM model, which focuses on generating high-quality medical reports.
+
+## Steps
+
+### 1. Download Datasets
+
+We are utilizing the MIMIC Chest X-ray JPG ([MIMIC-CXR-JPG](https://physionet.org/content/mimic-cxr-jpg/2.1.0/)) Database v2.0.0 for the task of medical report generation. This publicly available dataset includes 377,110 chest radiograph images in JPG format, along with structured labels derived from 227,827 free-text radiology reports. Derived from MIMIC-CXR, it provides a processed version of the original DICOM images, with standardized data splits and image labels. The dataset is de-identified in accordance with HIPAA Safe Harbor requirements, ensuring the removal of protected health information (PHI), and is designed to facilitate research in medical image analysis, natural language processing, and decision support.
+To refine the quality of the reports and eliminate noise, we utilize an enhanced text version developed by [DCL](https://github.com/mlii0117/DCL), and subsequently apply additional cleansing procedures to further optimize report accuracy.
+
+### 2. Convert Text Using LLM
+
+This step involves two sub-steps where the LLM helps in processing the textual reports to improve the quality of the data used for **medical report generation**.
+We are utilizing large language models (LLMs), such as Llama 3.1, to process our text data. Users can obtain an API key for NVIDIA [NIM](https://build.nvidia.com/explore/discover) to run scripts using the [Llama-3.1-8b-instruct](https://build.nvidia.com/meta/llama-3_1-8b-instruct) or [Llama-3.1-70b-instruct](https://build.nvidia.com/meta/llama-3_1-70b-instruct) models.
+
+#### 2.1 Collect a Sample Sentence Pool
+
+Before performing the text conversion, the LLM will be used to analyze the dataset and collect a sample pool of sentences. This pool will consist of commonly occurring phrases or structures found in the medical reports. These standardized phrases guide the text transformation process to ensure consistent input for report generation.
+
+```bash
+python 2-1_collect-sentence-pool.py
+```
+
+#### 2.2 Use LLM and Sentence Pool to Convert Text
+
+In this step, the LLM uses the previously collected sample sentence pool to convert the original medical report text into a standardized format. The LLM will replace or reformat certain sentences and medical terminology to ensure the text is uniform and ready for training the VLM model to generate reliable and coherent medical reports.
```

Comment:
Please describe the current implementation uses `openai` to call the LLMs. Developers may use their own locally-deployed ones.

Also, I think you will need to define the `base_url` somewhere, but I couldn't find it. Is the following necessary to call the API?

```
client = OpenAI(
  base_url = "https://integrate.api.nvidia.com/v1",
  api_key = ...
)
```

Reply:
Thanks. I have updated the scripts.